src/Symfony/Component/TypeInfo/TypeResolver/ChainTypeResolver.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo\TypeResolver;
+
+use Symfony\Component\TypeInfo\Exception\UnsupportedException;
+use Symfony\Component\TypeInfo\Type;
+use Symfony\Component\TypeInfo\TypeContext\TypeContext;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ * @author Baptiste Leduc <baptiste.leduc@gmail.com>
+ */
+final readonly class ChainTypeResolver implements TypeResolverInterface
+{
+    /**
+     * @param iterable<TypeResolverInterface> $typeResolvers
+     */
+    public function __construct(
+        private iterable $typeResolvers,
```

Comment:
I'd like to challenge the priority-based resolver list here as I'm wondering if it's really appropriate.
Couldn't this parameter be given a map of resolvers keyed by "resolver type" instead? e.g.
```php
ServiceLocator([
    \ReflectionParameter::class => fn () =>  $this->container->get('type_info.resolver.reflection_parameter'),
    \ReflectionProperty::class => fn () =>  $this->container->get('type_info.resolver.reflection_property'),
    'string' => fn () => ....,
    // ...
])
```
And this "entrypoint" service be responsible for finding the right resolver based on the `$subject instanceof X` dance?

Reply:
I really like the idea! It'll be a bit less extensible, but I don't think that point is the important one. By doing that, it'll be way more predictable.

I'd create an "entrypoint" service so that we'll be able to add custom logic at some point if it's necessary.